### DISCUSSION TITLE
Discord bot integration

### DISCUSSION THREAD

**Discussion by christo-achkov** [oobabooga/text-generation-webui#996] [Ideas]
This one should be relatively simple to implement and wil provide most users with an easy way to use the bot when they are not on their PC.  I think a simple non group 1 on 1 chat support would be a good step.

**Comment by TheMeIonGod** [comment]
There is this: https://github.com/xNul/chat-llama-discord-bot
But It is very simple. Like I would like to see the ability to use characters and stuff. It would be cool if something similar was a native module in text-generation-webui though.

**Reply by mercm8** [reply]
I've been messing around with this one a bit, trying to get it to load characters via Chat.load_character() but it doesnt seem to work correctly, as if the example dialogue isnt being fed into the model or something. 

If I knew more about this git stuff I'd try one of those branches or pull requests or whatever it's called. But right now it's just a bunch of spaghetti on the wall.
![image](https://user-images.githubusercontent.com/37743453/232236867-c0f539ce-38df-416c-b3f0-8b71dbc77075.png)

update, here's my fork: https://github.com/mercm8/chat-llama-discord-bot

**Comment by SvetZitrka** [comment]
any news?

**Reply by TheMeIonGod** [reply]
This is pretty basic too but this works as an extension. It still needs a lot of work like being able to reply on discord. https://github.com/ChobPT/text-generation-webui-discord_bot

**Reply by sarutobiumon** [reply]
@oobabooga another extension is born

**Reply by SvetZitrka** [reply]
Any reference?

**Comment by chrisrude** [comment]
I also threw this together, though it runs as a separate process, not an extension.  Perhaps I could extension-ify it if there was interest?

https://github.com/chrisrude/oobabot

**Reply by TruthSearchers** [reply]
> I also threw this together, though it runs as a separate process, not an extension. Perhaps I could extension-ify it if there was interest?
> 
> https://github.com/chrisrude/oobabot

Thats cool

**Reply by Skrownerve** [reply]
This is awesome and I really hope you continue to develop it. Exactly what I've been looking for/hopelessly pondering how to build. What's the best way to pass along ideas/requests for features?

**Reply by chrisrude** [reply]
Happy to hear it!  I'd love any feedback anyone has.  Just create an issue on the project's github page:
https://github.com/chrisrude/oobabot/issues

It's something I'm actively working on, so suggestions are most welcome.

**Comment by erazor83** [comment]
Wow, I was just about to write one myself. However I wanted to use SillyTavern as a backend but they don't have an external API and it seems the developers aren't interested in writing one.

**Reply by TimyIsCool** [reply]
do you have any that runs on sillytavern?

**Reply by erazor83** [reply]
I was asking the dev about some kind of external API but he made pretty clear this will not be implemented since sillytavern should be treated as a fronted only. Hovever, i think it can be easily done via a sillytavern plugin. I'm not much of a nodejs developer so that's nothing for me.
